flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestTableProperties.java

```diff
@@ -0,0 +1,197 @@
+  public static Iterable<Object[]> parameters() {
+    List<Object[]> parameters = Lists.newArrayList();
+    for (FileFormat format : new FileFormat[] {FileFormat.ORC, FileFormat.AVRO, FileFormat.PARQUET}) {
+      for (Boolean isStreaming : new Boolean[] {true, false}) {
+        for (Object[] catalogParams : FlinkCatalogTestBase.parameters()) {
+          String catalogName = (String) catalogParams[0];
+          Namespace baseNamespace = (Namespace) catalogParams[1];
+          parameters.add(new Object[] {catalogName, baseNamespace, format, isStreaming});
+        }
+      }
+    }
+    return parameters;
+  }
+
+  @Override
+  protected TableEnvironment getTableEnv() {
+    if (tEnv == null) {
+      synchronized (this) {
+        EnvironmentSettings.Builder settingsBuilder = EnvironmentSettings
+            .newInstance();
+        if (isStreamingJob) {
+          settingsBuilder.inStreamingMode();
+          StreamExecutionEnvironment env = StreamExecutionEnvironment
+              .getExecutionEnvironment(MiniClusterResource.DISABLE_CLASSLOADER_CHECK_CONFIG);
+          env.enableCheckpointing(400);
+          env.setMaxParallelism(2);
+          env.setParallelism(2);
+          tEnv = StreamTableEnvironment.create(env, settingsBuilder.build());
+        } else {
+          settingsBuilder.inBatchMode();
+          tEnv = TableEnvironment.create(settingsBuilder.build());
+        }
+      }
+    }
+    return tEnv;
+  }
+
+  @Override
+  @Before
+  public void before() {
+    super.before();
+    sql("CREATE DATABASE %s", flinkDatabase);
+    sql("USE CATALOG %s", catalogName);
+    sql("USE %s", DATABASE);
+  }
+
+  @Override
+  @After
+  public void clean() {
+    sql("DROP DATABASE IF EXISTS %s", flinkDatabase);
+    super.clean();
+  }
+
+  @Test
+  public void testParquetTableProperties() throws Exception {
+    if (format != FileFormat.PARQUET) {
+      return;
+    }
+    String parquetTableName = "test_table_parquet";
+
+    String groupSizeBytes = "10000";
+    String pageSizeBytes = "10000";
+    String dictSizeBytes = "10000";
+    String compressionCodec = "uncompressed";
+
+    sql("CREATE TABLE %s (id int, data varchar) with (" +
+            "'" + TableProperties.PARQUET_ROW_GROUP_SIZE_BYTES + "'='" + groupSizeBytes + "'," +
+            "'" + TableProperties.PARQUET_PAGE_SIZE_BYTES + "'='" + pageSizeBytes + "'," +
+            "'" + TableProperties.PARQUET_DICT_SIZE_BYTES + "'='" + dictSizeBytes + "'," +
+            "'" + TableProperties.PARQUET_COMPRESSION + "'='" + compressionCodec + "'," +
+            "'" + TableProperties.DEFAULT_FILE_FORMAT + "'='%s')",
+        parquetTableName, format);
+    Table icebergTable = validationCatalog.loadTable(TableIdentifier.of(icebergNamespace, parquetTableName));
+
+    Parquet.WriteBuilder writeBuilder = Mockito.spy(Parquet.write(Files.localOutput(TEMPORARY_FOLDER.newFile())));
+
+    writeBuilder.forTable(icebergTable);
```

Comment:
I don't think it is necessary to have a suite for Flink. All this does is the same thing that the Parquet and Avro tests do, which is to validate that `forTable` calls `setAll` with table properties. There is no benefit to creating a table through Flink vs creating a table directly. Can you remove this Flink test?

Reply:
Deleted, looks like I need to do more research. 😄